### ISSUE TITLE
INVALID_PARAMETER_VALUE: Changing param values is not allowed

### ISSUE THREAD

**Issue by EvanZ** [Lightning-AI/pytorch-lightning#17830]
### Bug description

I'm  in the process of upgrading from 1.9.4 to 2.0.3 and finding that the following error is coming up during `on_train_start` where the only line of code is `self.logger.log_hyperparams(self.hparams)`. Here's the full error:

```
Exception has occurred: RestException
INVALID_PARAMETER_VALUE: Changing param values is not allowed. Params were already logged='[{'key': 'lr', 'old_value': '0.0001', 'new_value': '1.9054607179632475e-07'}]' for run ID='c2a50c03d7514612936b1be8f735971f'.
  File "/home/ec2-user/***/src/model/model.py", line 546, in on_train_start
    self.logger.log_hyperparams(self.hparams)
  File "/home/ec2-user/***/finetune_script.py", line 192, in <module>
    trainer.fit(model, datamodule=dm)
mlflow.exceptions.RestException: INVALID_PARAMETER_VALUE: Changing param values is not allowed. Params were already logged='[{'key': 'lr', 'old_value': '0.0001', 'new_value': '1.9054607179632475e-07'}]' for run ID='c2a50c03d7514612936b1be8f735971f'.
```

I should note that previously (when using v.1.9.4) I was using `auto_lr_find=True` in the Trainer object. But now I am using the `LearningRateFinder` callback. I can see in the log output that seems to be working(?):

```
Finding best initial lr:   0%|                                                                 | 0/100 [00:00<?, ?it/s]/home/ec2-user/.env/lib64/python3.8/site-packages/torch/optim/lr_scheduler.py:138: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  warnings.warn("Detected call of `lr_scheduler.step()` before `optimizer.step()`. "
Finding best initial lr:  99%|███████████████████████████████████████████████████████▍| 99/100 [00:06<00:00, 16.40it/s]`Trainer.fit` stopped: `max_steps=100` reached.
Finding best initial lr: 100%|███████████████████████████████████████████████████████| 100/100 [00:06<00:00, 15.11it/s]
Learning rate set to 1.9054607179632475e-07
```

I'm using the `MLFlowLogger` from `lightning.pytorch.loggers.mlflow`.

### What version are you seeing the problem on?

v2.0

### Error messages and logs

```
Exception has occurred: RestException
INVALID_PARAMETER_VALUE: Changing param values is not allowed. Params were already logged='[{'key': 'lr', 'old_value': '0.0001', 'new_value': '1.9054607179632475e-07'}]' for run ID='c2a50c03d7514612936b1be8f735971f'.
  File "/home/ec2-user/deep-behavior-embedding/src/model/lightning_model.py", line 546, in on_train_start
    self.logger.log_hyperparams(self.hparams)
  File "/home/ec2-user/deep-behavior-embedding/finetune_tweedie_iterable.py", line 192, in <module>
    trainer.fit(model, datamodule=dm)
mlflow.exceptions.RestException: INVALID_PARAMETER_VALUE: Changing param values is not allowed. Params were already logged='[{'key': 'lr', 'old_value': '0.0001', 'new_value': '1.9054607179632475e-07'}]' for run ID='c2a50c03d7514612936b1be8f735971f'.
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0): 2.0.3
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 2.0): 1.13
#- Python version (e.g., 3.9): 3.8.16
#- OS (e.g., Linux): Linux
#- CUDA/cuDNN version: 11.6
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source): pip
#- Running environment of LightningApp (e.g. local, cloud): EC2
```

</details>

cc @awaelchli @borda @Blaizzy

**Comment by SkafteNicki** [comment]
Hi, thanks for raising this issue. While this is quite old at this point I am going to answer nonetheless:

This has to do with a limitation with `mlflow` namely that a parameter that has already been logged cannot be logged once again. This is not something that lightning controls. It happens because lightning will automatically log all your models hyperparameters (I assume you call `save_hyperparameters` in the `__init__` of your `LightningModule`). This trainer arg control the behavior:
https://github.com/Lightning-AI/pytorch-lightning/blob/b7ca4d365978512ae053d7d8a8c2ddc99894e813/src/lightning/pytorch/trainer/trainer.py#L306-L307
So the issue is really a configuration/usage issue. The strategy to solve this is then to disable automatic hyperparameter logging: `Trainer(enable_autolog_hparams=False, ...)`  and then manually call `self.logger.log_hyperparameters` after the learning rate finder is done. An alternative is to use the tuner interface with temp trainer with logging disabled:

```python
trainer = Trainer(logger=False)
tuner = Tuner(trainer)
tuner.lr_find(..., update_attr=True)
trainer = Trainer(logger=MLFlowLogger(...), ...)
trainer.fit(...)
```

Closing issue, but feel free to ping and reopen if necessary.
